test-framework/junit5-component/src/main/java/io/quarkus/test/component/TestConfigProperty.java

```diff
@@ -1,22 +1,30 @@
 package io.quarkus.test.component;
 
+import static java.lang.annotation.ElementType.METHOD;
 import static java.lang.annotation.ElementType.TYPE;
 import static java.lang.annotation.RetentionPolicy.RUNTIME;
 
 import java.lang.annotation.Repeatable;
 import java.lang.annotation.Retention;
 import java.lang.annotation.Target;
 
+import org.junit.jupiter.api.TestInstance.Lifecycle;
+
 import io.quarkus.test.component.TestConfigProperty.TestConfigProperties;
 
 /**
- * Set the value of a configuration property for a {@code io.quarkus.test.component.QuarkusComponentTest}.
+ * Set the value of a configuration property.
+ * <p>
+ * If declared on a class then the configuration property is used for all test methods declared on a test class.
+ * <p>
+ * If declared on a method then it should be only used for a specific test method. Keep in mind that the annotation will be
+ * ignored if {@link Lifecycle#PER_CLASS} is used.
```

Comment:
The wording confuses me a little. Does this mean that with `Lifecycle.PER_CLASS`, this annotation cannot be used (= is ignored) on methods? If so, I'd write it like this:

> If declared on a method then the configuration property is only used for that test method. If the test lifecycle is {@link Lifecycle#_PER_CLASS}, this annotation can only be used on the test class and is ignored on test methods.

Reply:
Yes. When `Lifecycle.PER_CLASS` is used we don't restart the container per test method and thus the config cannot be updated. I will use your wording ;-)